File: core/src/com/jupiter/europa/entity/effects/BaseEffect.java
package com.jupiter.europa.entity.effects;

import com.badlogic.ashley.core.Entity;
import com.badlogic.gdx.utils.Json;
import com.badlogic.gdx.utils.JsonValue;

/**
 * Created by nathan on 5/21/15.
 */
public class BaseEffect implements Effect {

    // Fields
    private Entity entity;


    // Properties
    public final Entity getEntity() {
        return this.entity;
    }


    // Public Methods
    @Override
    public void onAdd(Entity entity) {
        this.entity = entity;
    }

    @Override
    public void onRemove() {
        this.entity = null;
    }

    @Override
    public void update(float deltaT) {

    }

    @Override
    public void onCombatTurnStart() {

    }

    @Override
    public void onCombatTurnEnd() {

    }

    @Override
    public void onOutOfCombatTurn() {

    }


    // Serializable Implementation
    @Override
    public void read(Json json, JsonValue jsonValue) {

    }

    @Override
    public void write(Json json) {

    }
}

File: core/src/com/jupiter/europa/entity/effects/BaseEffect.kt
/*
 * The MIT License
 *
 * Copyright 2015 Nathan Templon.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 *
 */

package com.jupiter.europa.entity.effects

import com.badlogic.ashley.core.Entity
import com.badlogic.gdx.utils.Json
import com.badlogic.gdx.utils.JsonValue

/**
 * Created by nathan on 5/21/15.
 */
public open class BaseEffect() : Effect {

    // Properties
    public var entity: Entity? = null
        private set


    // Public Methods
    override fun onAdd(entity: Entity) {
        this.entity = entity
    }

    override fun onRemove() {
        this.entity = null
    }

    override fun update(deltaT: Float) {
    }

    override fun onCombatTurnStart() {
    }

    override fun onCombatTurnEnd() {
    }

    override fun onOutOfCombatTurn() {
    }


    // Serializable Implementation
    override fun read(json: Json, jsonValue: JsonValue) {
    }

    override fun write(json: Json) {

    }
}
